### PR TITLE
Handle optimistic response for reactions

### PR DESCRIPTION
**Pull request by hochiw**
- Use regular gql query with skip condition instead of lazy query
- Fix type policy handling for proposal objects
-  Update reaction badge active colour according to design
- Handle proposal reaction optimistic response

refs oursky/likedao#224

### CODE REVIEW

File: react-app/src/api/reactionAPI.ts [hunk 1]
```diff
@@ -32,27 +116,51 @@ export const useReactionAPI = (): IReactionAPI => {
...
             targetId,
             reaction,
           },
         },
+        // Optimistic response to trigger update function
+        optimisticResponse: {
+          __typename: "Mutation",
+          setReaction: {
+            __typename: "Reaction",
+            id: "",
```
Comment (chihimng) [inline react-app/src/api/reactionAPI.ts:136]: will assigning static dummy id affect caching?

Comment (chihimng) [reply react-app/src/api/reactionAPI.ts:136]: or will this be discarded after update hook is invoked?

**Comment by hochiw** [comment]
@chihimng updated 🙏

**Review by elise-ng — APPROVED** [review]

**Comment by chihimng** [comment]
any way for us to fetch the true value from server after successful mutation?

**Comment by chihimng** [comment]
or alternatively, do we revert the optimistic result if we receive error currently?

**Comment by hochiw** [comment]
let me see see

**Comment by hochiw** [comment]
@chihimng  

> any way for us to fetch the true value from server after successful mutation?

I have changed to using the reaction value provided by the update function instead of user input, the flow is now as follows

user press reaction -> sends request to gql server -> generates optimistic response -> update function called with optimistic response -> update ui -> response returned from server -> update function called with actual data -> update ui

> alternatively, do we revert the optimistic result if we receive error currently?

yes it rolls back automatically, tested by throwing error on server side

**Review by elise-ng — APPROVED** [review]

**Pull request merged**